packages/paypal-commerce-integration/src/paypal-commerce-alternative-methods/paypal-commerce-alternative-method-ratepay-payment-strategy.ts

```diff
@@ -0,0 +1,233 @@
+import {
+    InvalidArgumentError,
+    OrderFinalizationNotRequiredError,
+    OrderRequestBody,
+    PaymentArgumentInvalidError,
+    PaymentInitializeOptions,
+    PaymentIntegrationService,
+    PaymentMethodInvalidError,
+    PaymentRequestOptions,
+    PaymentStrategy,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+import { LoadingIndicator } from '@bigcommerce/checkout-sdk/ui';
+
+import PayPalCommerceIntegrationService from '../paypal-commerce-integration-service';
+
+import { WithPayPalCommerceAlternativeMethodsPaymentInitializeOptions } from '../index';
+import { BirthDate, PayPalCommerceInitializationData } from '../paypal-commerce-types';
+import { PaypalCommerceRatePay } from './paypal-commerce-alternative-methods-payment-initialize-options';
+
+export default class PayPalCommerceAlternativeMethodRatePayPaymentStrategy
+    implements PaymentStrategy
+{
+    private loadingIndicatorContainer?: string;
+    private guid?: string;
+    private paypalcommerceratepay?: PaypalCommerceRatePay;
+    private orderId?: string;
+    constructor(
+        private paymentIntegrationService: PaymentIntegrationService,
+        private paypalCommerceIntegrationService: PayPalCommerceIntegrationService,
+        private loadingIndicator: LoadingIndicator,
+    ) {}
+
+    async initialize(
+        options: PaymentInitializeOptions &
+            WithPayPalCommerceAlternativeMethodsPaymentInitializeOptions,
+    ): Promise<void> {
+        const { gatewayId, methodId, paypalcommerceratepay } = options;
+
+        if (!methodId) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "options.methodId" argument is not provided.',
+            );
+        }
+
+        if (!gatewayId) {
+            throw new InvalidArgumentError(
+                'Unable to initialize payment because "options.gatewayId" argument is not provided.',
+            );
+        }
+
+        if (!paypalcommerceratepay) {
+            throw new InvalidArgumentError(
+                `Unable to initialize payment because "options.paypalcommerceratepay" argument is not provided.`,
+            );
+        }
+
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow<PayPalCommerceInitializationData>(
+            methodId,
+            gatewayId,
+        );
+
+        const { orderId } = paymentMethod.initializationData || {};
+        this.paypalcommerceratepay = paypalcommerceratepay;
+
+        if (orderId) {
+            this.orderId = orderId;
+            return;
+        }
```

Comment:
I don't really sure that this code should work for ratepay

Reply:
removed